src/pages/[platform]/build-a-backend/functions/examples/custom-auth-flows/index.mdx

```diff
@@ -0,0 +1,215 @@
+
+To get started, create the first of the three triggers, `create-auth-challenge`. This is the trigger responsible for creating the reCAPTCHA challenge after a password is verified.
+
+```ts title="amplify/auth/create-auth-challenge/resource.ts"
+import { defineFunction } from "@aws-amplify/backend"
+
+export const createAuthChallenge = defineFunction({
+  name: "create-auth-challenge",
+})
+```
+
+After creating the resource file, create the handler with the following contents:
+
+```ts title="amplify/auth/create-auth-challenge/handler.ts"
+import type { CreateAuthChallengeTriggerHandler } from "aws-lambda";
+
+export const handler: CreateAuthChallengeTriggerHandler = async (event) => {
+  if (event.request.challengeName === "CUSTOM_CHALLENGE") {
+    // Generate a random code for the custom challenge
+    const challengeCode = "123456";
+
+    event.response.challengeMetadata = "TOKEN_CHECK";
+
+    event.response.publicChallengeParameters = {
+      trigger: "true",
+      code: challengeCode,
+    };
+
+    event.response.privateChallengeParameters = { trigger: "true" };
+    event.response.privateChallengeParameters.answer = challengeCode;
+  }
+  return event;
+};
+```
+
+## Define auth challenge trigger
+
+Next, you will want to create the trigger responsible for _defining_ the auth challenge flow, `define-auth-challenge`.
+
+```ts title="amplify/auth/define-auth-challenge/resource.ts"
+import { defineFunction } from "@aws-amplify/backend"
+
+export const defineAuthChallenge = defineFunction({
+  name: "define-auth-challenge",
+})
+```
+
+After creating the resource file, create the handler with the following contents if you are using `CUSTOM_WITHOUT_SRP`:
+
+```ts title="amplify/auth/define-auth-challenge/handler.ts"
+import type { DefineAuthChallengeTriggerHandler } from "aws-lambda"
+
+export const handler: DefineAuthChallengeTriggerHandler = async (event) => {
+  if (event.request.session.length === 0) {
+    // If it's the first auth stage
+    event.response.issueTokens = false;
+    event.response.failAuthentication = false;
+    event.response.challengeName = "CUSTOM_CHALLENGE";
+  } else if (
+    event.request.session.length === 1 &&
+    event.request.session[0].challengeName === "CUSTOM_CHALLENGE" &&
+    event.request.session[0].challengeResult === true
+  ) {
+    // If CUSTOM_CHALLENGE is passed
+    event.response.issueTokens = true;
+    event.response.failAuthentication = false;
+  } else {
+    // Fail auth if we don't have the expected challenge results
+    event.response.issueTokens = false;
+    event.response.failAuthentication = true;
+  }
+
+  return event;
+};
+```
+
+Or if you are using `CUSTOM_WITH_SRP`:
+
+```ts title="amplify/auth/define-auth-challenge/handler.ts"
+import type { DefineAuthChallengeTriggerHandler } from "aws-lambda"
+
+export const handler: DefineAuthChallengeTriggerHandler = async (event) => {
+  if (event.request.session.length === 0) {
+    event.response.issueTokens = false;
```

Comment:
for folks seeing this trigger for the first time it's tough to grok what each of these steps mean or are doing, can we add a short comment like "verify password challenge" inside each of the conditional blocks?